src/sagemaker_containers/encoders.py

```diff
@@ -38,17 +38,36 @@ def array_to_npy(array_like):  # type: (np.array or Iterable or int or float) ->
     return buffer.getvalue()
 
 
-def npy_to_numpy(npy_array):  # type: (object) -> np.array
+def npy_to_numpy(npy_array, dtype=np.float32):  # type: (object) -> np.array
     """Convert an NPY array into numpy.
 
     Args:
+        dtype (dtype, optional):  Data type of the resulting array.
+                                    If None, the dtypes will be determined by the contents of each column, individually.
         npy_array (npy array): to be converted to numpy array
 
     Returns:
         (np.array): converted numpy array.
     """
     stream = BytesIO(npy_array)
-    return np.load(stream)
+    return np.load(stream).astype(dtype)
```

Comment:
I don't think we should do `astype(dtype)`, at least not with `np.float32` instead of `none`. The dtype is serialized with the data -- we should default to that.

Reply:
+1